mlflow/tracing/fluent.py

```diff
@@ -260,7 +279,184 @@ def pagination_wrapper_func(number_to_get, next_page_token):
 
     get_display_handler().display_traces(results)
 
-    return results
+    traces_df = traces_to_df(results)
+    if extract_fields:
+        traces_df = extract(
+            traces=traces_df,
+            fields=extract_fields,
+            col_name=SPANS_COLUMN_NAME,
+        )
+
+    return traces_df
+
+
+def extract(
+    traces: Union[List["mlflow.entities.Trace"], "pandas.DataFrame"],
+    fields: List[str],
+    col_name: Optional[str] = None,
+) -> "pandas.DataFrame":
+    """
+    Extracts the specified fields from the spans contained in the specified traces.
+
+    Args:
+        traces: A list of MLflow Traces or a pandas DataFrame containing traces.
+        fields: A list of field strings of the form 'span_name.[inputs|outputs]' or
+            'span_name.[inputs|outputs].field_name'.
```

Comment:
Does this assume any format for span inputs/outputs? For example, the span inputs can be a single scaler e.g. integer, not always dictionary? Probably we want to add some validation and add examples in the docstring.

Reply:
We added test cases when the span input is not a dictionary. I think in that case, we would have to use `span_name.[inputs|outputs]` and won't be able to use `span_name.[inputs|outputs].field_name`

Reply:
Got it, silently returning empty result makes seese to me . Can we also add test case for other edge cases? For example, I guess span name can be any string, e.g including period like "my.llm".